src/frame/headers.rs

```diff
@@ -440,10 +440,14 @@ impl Pseudo {
     pub fn request(method: Method, uri: Uri) -> Self {
         let parts = uri::Parts::from(uri);
 
-        let path = parts
+        let mut path = parts
             .path_and_query
             .map(|v| v.into())
-            .unwrap_or_else(|| Bytes::from_static(b"/"));
+            .unwrap_or_else(|| Bytes::new());
+
+        if path.is_empty() && method != Method::OPTIONS {
```

Comment:
When the URI is "*" it isn't clear to me what `path` is supposed to be here. is it "*" like `Uri::path()` says or is it ""?

When URI is "*" and the method isn't OPTIONS then it seems like we should return an error somewhere here, but that doesn't seem to be done. Or maybe I'm misunderstanding some details of how "*" is handled in the parsing code?

Reply:
The relevant [spec section](http://httpwg.org/specs/rfc7540.html#rfc.section.8.1.2.3):

> The :path pseudo-header field includes the path and query parts of the target URI (the path-absolute production and optionally a '?' character followed by the query production (see Sections 3.3 and 3.4 of [RFC3986]). A request in asterisk form includes the value '*' for the :path pseudo-header field.
>
> This pseudo-header field MUST NOT be empty for http or https URIs; http or https URIs that do not contain a path component MUST include a value of '/'. The exception to this rule is an OPTIONS request for an http or https URI that does not include a path component; these MUST include a :path pseudo-header field with a value of '*' (see [RFC7230], Section 5.3.4).

Does this make sense?

I also saw you opened https://github.com/hyperium/http/issues/176, so we can discuss the details of the `http` path API there.

Reply:
So, the path can also be not set of the scheme is not `http` or `https`, but those cases are not really supported yet.

Reply:
I think this is fine, but it should also check for `Method::CONNECT.

[RFC7450#CONNECT](http://httpwg.org/specs/rfc7540.html#CONNECT):

> The :scheme and :path pseudo-header fields MUST be omitted.